contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -8362,6 +8375,84 @@ rewrite_function_trim_to_sys_trim(TSqlParser::TRIMContext *ctx)
 	rewritten_query_fragment.emplace(std::make_pair(ctx->TRIM()->getSymbol()->getStartIndex(), std::make_pair(::getFullText(ctx->TRIM()), "sys.trim")));
 }
 
+/*
+ * In this helper function we Rewrite the Query for XML Handling
+ * For Func_Ref Functions with args (such as EXIST(arg)) : ColRef.Func_name(arg_list)  ->  Func_name(arg_list, ColRef)
+ */
+template<class T>
+void
+rewrite_xml_func_query_helper(T ctx, TSqlParser::Method_callContext *method, size_t xml_start_index)
```

Comment:
We have done similar handling in case of spatial function, we can have a common function and reuse it for both the datatypes rather than code duplication

Reply:
Done